### PR TITLE
Fix repo owner filter

### LINKED ISSUE #2784: WOODPECKER_REPO_OWNERS is not working
### Component

server

### Describe the bug

I am looking for a way to limit which orgs can be activated in Woodpecker. According to the description in the documentation, `WOODPECKER_REPO_OWNERS` should do this, but it doesn't work. Even if `WOODPECKER_REPO_OWNERS=myorg` is set, I can still enable repos of `myorg` and `otherorg` in Woodpecker.

Looking at the code, it looks like the variable is no longer used:

```
❯ grep -ri "OwnersWhitelist"
server/model/settings.go:	OwnersWhitelist map[string]bool // Owners whitelist
server/router/middleware/config.go:		OwnersWhitelist: sliceToMap2(c.StringSlice("repo-owners")),
```

Based on the Git history, it looks like the code was removed in https://github.com/woodpecker-ci/woodpecker/pull/1658

```
❯ git log -SOwnersWhitelist
commit 0970f35df5034633a40b8d9f1a2ab901963aebee
Author: qwerty287 <80460567+qwerty287@users.noreply.github.com>
Date:   Tue Mar 21 23:01:59 2023 +0100

    Do not store inactive repos (#1658)
    
    Do not sync repos with forge if the repo is not necessary in DB.
    
    In the DB, only repos that were active once or repos that are currently
    active are stored. When trying to enable new repos, the repos list is
    fetched from the forge instead and displayed directly. In addition to
    this, the forge func `Perm` was removed and is now merged with `Repo`.
    
    Solves a TODO on RepoBatch.
    
    ---------
    
    Co-authored-by: Anbraten <anton@ju60.de>
```

Was it removed by accident or is there now another way to limit organizations in Woodpecker and `WOODPECKER_REPO_OWNERS` is just a leftover?

\cc @qwerty287 

### System Info

```shell
next
```

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

### PR DESCRIPTION
**Pull request by qwerty287**
and move to server config instead of middleware

cc @xoxys 

closes #2784

### CODE REVIEW

File: server/plugins/permissions/util.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package permissions
+
+// sliceToMap is a helper function to convert a string slice to a map.
+func sliceToMap(s []string) map[string]bool {
```
Comment (6543) [inline server/plugins/permissions/util.go:4]: could you move and make it similar to
https://github.com/woodpecker-ci/woodpecker/blob/905bb9585371b3e86393d994f31983d936e24a91/shared/utils/slices.go#L53

?

**Review by xoxys — COMMENTED** [review]
As we touch it anyway can we call it allowlist instead of whitelist?

**Review by xoxys — APPROVED** [review]
Thanks for the fix. I like the new permissions struct, makes the code more readable.

**Comment by pat-s** [comment]
A partly unrelated thought: could we apply a similar approach to solve https://github.com/woodpecker-ci/woodpecker/issues/336?

**Pull request merged**